Userland/Libraries/LibCore/System.cpp

```diff
@@ -287,4 +287,17 @@ ErrorOr<struct passwd*> getpwnam(StringView name)
         return Error::from_syscall("getpwnam"sv, -errno);
 }
 
+ErrorOr<struct group*> getgrnam(StringView name)
+{
+    ::setgrent();
+    while (auto* gr = ::getgrent()) {
+        if (gr->gr_name == name)
+            return gr;
+    }
+    if (errno == 0)
+        return nullptr;
```

Comment:
I'm honestly not sure how this should work? The caller interprets null as error... And the normal pattern is that try returns an unwrapped value that is know to be valid. So returning a nullptr here instead of an error is kind is confusing. But... It's also how the real API works. Maybe others have  more clarity on what we should do here?

Reply:
That's more or less how I reasoned about this as well. Didn't feel good, but unsure how intrusive it would become if it returns an error.

Reply:
Would changing `getpwnam` and `getgrnam` to `Error::from_string_literal("getpwnam: Unknown username"sv);` if it doesn't find the username or group name.

```cpp
ErrorOr<struct passwd*> getpwnam(StringView name)
{
    ::setpwent();
    while (auto* pw = ::getpwent()) {
        if (pw->pw_name == name)
            return pw;
    }
    if (errno == 0)
        Error::from_string_literal("getpwnam: Unknown username"sv);
    else
        return Error::from_syscall("getpwnam"sv, -errno);
}

ErrorOr<struct group*> getgrnam(StringView name)
{
    ::setgrent();
    while (auto* gr = ::getgrent()) {
        if (gr->gr_name == name)
            return gr;
    }
    if (errno == 0)
        Error::from_string_literal("getgrnam: Unknown group name"sv);
    else
        return Error::from_syscall("getgrnam"sv, -errno);
}
```